src/main/kotlin/org/jetbrains/research/testspark/tools/llm/Llm.kt

```diff
@@ -35,11 +36,11 @@ class Llm(override val name: String = "LLM") : Tool {
     private fun getLLMProcessManager(project: Project, psiFile: PsiFile, caretOffset: Int, testSamplesCode: String): LLMProcessManager {
         val classesToTest = mutableListOf<PsiClass>()
 
-        ApplicationManager.getApplication().runReadAction(
-            Computable {
+        ProgressManager.getInstance().runProcessWithProgressSynchronously({
+            ApplicationManager.getApplication().runReadAction {
```

Comment:
If it runs synchronously, does it block the main thread during the execution (as it was before for test compilation)?